app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/LecturesByRoomTest.kt

```diff
@@ -0,0 +1,75 @@
+package nerd.tuxmobil.fahrplan.congress.schedule
+
+import com.google.common.truth.Truth.assertThat
+import nerd.tuxmobil.fahrplan.congress.models.Lecture
+import org.junit.Test
+
+class LecturesByRoomTest {
+
+    @Test
+    fun `no lectures`() {
+        val lecturesByRoom = LecturesByRoom(emptyList())
+
+        assertThat(lecturesByRoom).isEmpty()
+    }
+
+    @Test
+    fun `one lecture in room zero`() {
+        val roomZeroLecture = Lecture("").apply { roomIndex = 0 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomZeroLecture))
+
+        assertThat(lecturesByRoom).hasSize(1)
+        assertThat(lecturesByRoom[0]).containsExactly(roomZeroLecture)
+    }
+
+    @Test
+    fun `one lecture in room index 1 and 3 - creates extract empty rooms for index 0 and 2`() {
+        val roomZeroLecture = Lecture("").apply { roomIndex = 1 }
+        val roomThreeLecture = Lecture("").apply { roomIndex = 3 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomZeroLecture, roomThreeLecture))
+
+        assertThat(lecturesByRoom).hasSize(4)
+        assertThat(lecturesByRoom[0]).isEmpty()
+        assertThat(lecturesByRoom[1]).containsExactly(roomZeroLecture)
+        assertThat(lecturesByRoom[2]).isEmpty()
+        assertThat(lecturesByRoom[3]).containsExactly(roomThreeLecture)
+    }
+
+    @Test
+    fun `one lecture per room - starting from zero, unordered`() {
+        val roomZeroLecture = Lecture("").apply { roomIndex = 0 }
+        val roomOneLecture = Lecture("").apply { roomIndex = 1 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomOneLecture, roomZeroLecture))
+
+        assertThat(lecturesByRoom).hasSize(2)
+        assertThat(lecturesByRoom[0]).containsExactly(roomZeroLecture)
+        assertThat(lecturesByRoom[1]).containsExactly(roomOneLecture)
+    }
+
+    @Test
+    fun `one lecture per room - starting from one, with sparse list`() {
+        val roomOneLecture = Lecture("").apply { roomIndex = 1 }
+        val roomThreeLecture = Lecture("").apply { roomIndex = 3 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomOneLecture, roomThreeLecture))
+
+        assertThat(lecturesByRoom).hasSize(4)
+        assertThat(lecturesByRoom[1]).containsExactly(roomOneLecture)
+        assertThat(lecturesByRoom[3]).containsExactly(roomThreeLecture)
+    }
```

Comment:
:mag: Why are two empty rooms created just by the fact that the room indices are not consecutive?

Reply:
I changed LecturesByRoom implementation to make it more robust / null safe. Rooms belonging to sparse indices are mapped to an empty list.

Reply:
Please add KDoc to the `LecturesByRoom` class which explains behavior of the class and the motivation behind that decision.